ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeStyle/FantomasOptionsPage.fs

```diff
@@ -0,0 +1,350 @@
+            .GetBeRichText() :> BeControl
+
+    let validate (comboBox: BeComboBox) (settingsData: Dictionary<_, _>) =
+        let settingName = settingsData.Keys |> Seq.sortBy id |> Seq.item comboBox.SelectedIndex.Value
+        match settingsData[settingName] with
+        | { Status = FailedToRun } ->
+            ValidationResult(ValidationStates.validationWarning, "The specified Fantomas version failed to run.")
+        | { Status = UnsupportedVersion } ->
+            ValidationResult(ValidationStates.validationWarning, $"Supported Fantomas versions: {MinimalSupportedVersion} and later.")
+        | { Status = SelectedButNotFound } ->
+            ValidationResult(ValidationStates.validationWarning, "The specified Fantomas version not found.")
+        | _ ->
+            ValidationResult(ValidationStates.validationPassed)
+
+    let createComboBox (key: JetBrains.DataFlow.IProperty<FantomasVersionSettings>) =
+        let settingsData = fantomasDetector.GetSettings()
+        let beComboBoxFromEnum =
+            key.GetBeComboBoxFromEnum(lifetime,
+                PresentComboItem (fun x y z -> formatSettingItem y settingsData),
+                seq { FantomasVersionSettings.SolutionDotnetTool
+                      FantomasVersionSettings.GlobalDotnetTool } |> Seq.filter (not << settingsData.ContainsKey)
+            )
+        let beComboBoxFromEnum = beComboBoxFromEnum.WithValidationRule(lifetime, (fun () ->
+            let res = validate beComboBoxFromEnum settingsData
+            struct(res.ResultMessage, res.State)))
+
+        let validationLabel = BeValidationLabel(BeControls.GetRichText())
+
+        let beSpanGrid = BeControls.GetSpanGrid("auto,auto")
+
+        beComboBoxFromEnum.ValidationResult.Change.Advise(lifetime, fun x -> validationLabel.ValidationResult.Value <- ValidationResult(x.State, x.ResultMessage) )
+        beComboBoxFromEnum.SelectedIndex.Change.Advise(lifetime, fun x -> beComboBoxFromEnum.Revalidate.Fire(ValidationTrigger.ValueChanged))
+        beComboBoxFromEnum.ValidationResult.Value <- validate beComboBoxFromEnum settingsData
+
+        beSpanGrid
+            .AddColumnElementsToNewRow(BeSizingType.Fit, true, [|"Fantomas version".GetBeLabel() :> BeControl; beComboBoxFromEnum.WithMinSize(BeControlSizeFixed(BeControlSizeType.FIT_TO_CONTENT, BeControlSizeType.FIT_TO_CONTENT), lifetime)|])
+            .AddColumnElementsToNewRow(BeSizingType.Fit, false, [|"".GetBeLabel() :> BeControl; validationLabel |])
+
+    do
+        use indent = this.Indent()
+
+        this.AddControl((fun (key: FSharpFantomasOptions) -> key.Version), createComboBox) |> ignore//.DisableDescendantsWhenDisabled(this.Lifetime)
+
+        this.AddCommentText("To use a specified Fantomas version, install it globally via 'dotnet tool install fantomas-tool'\n" +
+                            "or specify it in dotnet-tools.json file in the solution directory. Supported Fantomas versions: 3.2 and later.")
```

Comment:
Any reason the version is inlined in some places and uses string interpolation in others?

Reply:
Btw, do we need to show the version here at all? Is it common enough to install such old versions via `dotnet`, so it's worth it to show it to every user opening this page?

Reply:
> Any reason the version is inlined in some places and uses string interpolation in others?

forgot about this place, thanks

> Btw, do we need to show the version here at all? Is it common enough to install such old versions via dotnet, so it's worth it to show it to every user opening this page?

yes, I think we can not follow the prototype here and really not show.